src/actions/contest_actions.js

```diff
@@ -14,6 +15,9 @@ export const receiveContests = payload => ({
 
 export const fetchContests = () => async dispatch => {
   const payload = await ContestAPIUtil.fetchContests();
+  Object.keys(payload.contests).forEach( contest_id => {
+    dispatch(fetchEntry(payload.contests[contest_id].winning_entry_id));
+  });
```

Comment:
This really does not seem ideal. When we have all 50 contests being fetched, that's 50 separate API calls to get each winning entry. Can we just include the winning entry for each contest in the jbuilder response to fetch all contests?

Reply:
I definitely like that way more. Will refactor and submit PRs.